assets/blocks/button/color-hooks.js

```diff
@@ -0,0 +1,124 @@
+import { addFilter } from '@wordpress/hooks';
+import { __ } from '@wordpress/i18n';
+import { omitBy } from 'lodash';
+import { withColorSettings } from '../../shared/blocks/settings';
+
+/**
+ * Check if block is a Sensei button.
+ *
+ * @param {Object|string} blockType Block settings or name.
+ * @return {boolean} Is Sensei button.
+ */
+const isSenseiButton = ( blockType ) => {
+	blockType = 'string' === typeof blockType ? blockType : blockType.name;
+	return blockType && blockType.startsWith( 'sensei-lms/button' );
+};
+
+/**
+ * Add fallback Color settings and attributes if the color support key is not available.
+ *
+ * @param {Object} settings Block settings.
+ * @return {Object} Block settings.
+ */
+export const addColorSettings = ( settings ) => {
+	if ( ! isSenseiButton( settings ) ) return settings;
+
+	if ( settings.supports?.color && ! settings.attributes?.backgroundColor ) {
+		settings.edit = withColorSettings( {
+			backgroundColor: {
+				style: 'background-color',
+				label: __( 'Background color', 'sensei-lms' ),
+			},
+			textColor: {
+				style: 'color',
+				label: __( 'Text color', 'sensei-lms' ),
+			},
+		} )( settings.edit );
+
+		settings.attributes = {
+			...settings.attributes,
+			backgroundColor: {
+				type: 'string',
+			},
+			customBackgroundColor: {
+				type: 'string',
+			},
+			textColor: {
+				type: 'string',
+			},
+			customTextColor: {
+				type: 'string',
+			},
+		};
+	}
+	return settings;
+};
+
+/**
+ * Remove colors from className and style props.
+ *
+ * @param {Object} props Block wrapper extra props.
+ * @return {Object} Block wrapper extra props.
+ */
+function removeColorProps( props ) {
+	return {
+		...props,
+		className: props.className?.replace(
+			/\s*has-[\w-]*(color|background)/g,
+			''
```

Comment:
A tiny problem with the color support hook is it adds the color classes and styles to the wrapper component both in the editor and on save. Since we want those on our button only, we remove them here.